crates/matrix-sdk/tests/integration/room/attachment/mod.rs

```diff
@@ -297,6 +303,328 @@ async fn test_room_attachment_send_mentions() {
+        .expect_mime_type("image/jpeg")
+        .ok(mxc_uri!("mxc://example.com/AQwafuaFswefuhsfAFAgsw"))
+        .mock_once()
+        .mount()
+        .await;
+
+    let client = mock.client_builder().build().await;
+    let room = mock.sync_joined_room(&client, &DEFAULT_TEST_ROOM_ID).await;
+    mock.mock_room_state_encryption().plain().mount().await;
+
+    let response = room
+        .send_attachment(
+            "image",
+            &mime::IMAGE_JPEG,
+            b"Hello world".to_vec(),
+            AttachmentConfig::new()
+                .mentions(Some(Mentions::with_user_ids([owned_user_id!("@user:localhost")])))
+                .reply(Some(Reply {
+                    event_id: replied_to_event_id.into(),
+                    enforce_thread: EnforceThread::Unthreaded,
+                })),
+        )
+        .await
+        .unwrap();
+
+    assert_eq!(expected_event_id, response.event_id);
+}
+
+#[async_test]
+async fn test_room_attachment_start_thread() {
+    let mock = MatrixMockServer::new().await;
+
+    let expected_event_id = event_id!("$h29iv0s8:example.com");
+    let replied_to_event_id = event_id!("$foo:bar.com");
+
+    mock.mock_room_send()
+        .body_matches_partial_json(json!({
+            "m.relates_to": {
+                "rel_type": "m.thread",
+                "event_id": replied_to_event_id,
+                "m.in_reply_to": {
+                    "event_id": replied_to_event_id
+                },
+                "is_falling_back": true
+            },
+        }))
+        .ok(expected_event_id)
+        .mock_once()
+        .mount()
+        .await;
+
+    let f = EventFactory::new();
+    mock.mock_room_event()
+        .match_event_id()
+        .ok(f
+            .text_msg("Send me your attachments")
+            .sender(*ALICE)
+            .event_id(replied_to_event_id)
+            .into())
+        .mock_once()
+        .mount()
+        .await;
+
+    mock.mock_upload()
+        .expect_mime_type("image/jpeg")
+        .ok(mxc_uri!("mxc://example.com/AQwafuaFswefuhsfAFAgsw"))
+        .mock_once()
+        .mount()
+        .await;
+
+    let client = mock.client_builder().build().await;
+    let room = mock.sync_joined_room(&client, &DEFAULT_TEST_ROOM_ID).await;
+    mock.mock_room_state_encryption().plain().mount().await;
+
+    let response = room
+        .send_attachment(
+            "image",
+            &mime::IMAGE_JPEG,
+            b"Hello world".to_vec(),
+            AttachmentConfig::new()
+                .mentions(Some(Mentions::with_user_ids([owned_user_id!("@user:localhost")])))
+                .reply(Some(Reply {
+                    event_id: replied_to_event_id.into(),
+                    enforce_thread: EnforceThread::Threaded(ReplyWithinThread::No),
+                })),
+        )
+        .await
+        .unwrap();
+
+    assert_eq!(expected_event_id, response.event_id);
+}
+
+#[async_test]
+async fn test_room_attachment_reply_on_thread() {
```

Comment:
I think this test is semantically equivalent to the `start_thread` one; maybe we can get rid of one of the twos, unless I've missed some important distinction between those?

Reply:
Yeah, you're right. It doesn't add much more. Have deleted it.